src/topology/continuous_function/compact.lean

```diff
@@ -394,4 +409,27 @@ end
 
 end comp_right
 
+section weierstrass
+variables {X : Type*} [topological_space X] [t2_space X] [locally_compact_space X]
+variables {E : Type*} [normed_group E] [complete_space E]
+
+/-- **Weierstrass M-test** -/
+lemma summable_of_locally_summable_norm {ι : Type*} {F : ι → C(X, E)}
+  (hF : ∀ K : set X, compact_space K → by exactI summable (λ i, ∥(F i).restrict K∥)) :
```

Comment:
Should we add `continuous_map.restrict_compact (f : C(X, Y)) (K : set X) (hK : is_compact K) : K →ᵇ Y` and use it here?